core/server/master/src/main/java/alluxio/master/file/FileSystemMasterClientRestServiceHandler.java

```diff
@@ -392,7 +393,8 @@ public Response remove(@QueryParam("path") final String path,
       @Override
       public Void call() throws Exception {
         Preconditions.checkNotNull(path, "required 'path' parameter is missing");
-        mFileSystemMaster.delete(new AlluxioURI(path), recursive);
+        mFileSystemMaster.delete(new AlluxioURI(path), DeleteOptions.defaults()
+            .setRecursive(recursive));
```

Comment:
we do not need to do anything for `PathsRestServiceHandler` because it uses the `DeleteOptions` directly in the API